tools/cp-caps/rcp_caps_test.py

```diff
@@ -136,9 +137,67 @@ def test_throughput(self):
 
         self.__output_format_string('Throughput', self.__bitrate_to_string(results['receiver']['bitrate']))
 
+    def test_link_metrics(self):
+        """Test whether the DUT supports Link Metrics Initiator and Subjector."""
+        self.__dataset = self.__get_default_dataset()
+
+        self.__dut.factory_reset()
+        self.__ref.factory_reset()
+
+        self.__dut.join(self.__dataset)
+        self.__dut.wait_for('state', 'leader')
+
+        self.__ref.join(self.__dataset)
+        self.__ref.wait_for('state', ['child', 'router'])
+
+        test_case = 'Link Metrics Initiator'
+        ref_linklocal_address = self.__ref.get_ipaddr_linklocal()
+        ret = self.__run_link_metrics_test_commands(initiator=self.__dut, subject_address=ref_linklocal_address)
+        self.__output_format_bool(test_case, ret)
+
+        test_case = 'Link Metrics Subject'
+        dut_linklocal_address = self.__dut.get_ipaddr_linklocal()
+        ret = self.__run_link_metrics_test_commands(initiator=self.__ref, subject_address=dut_linklocal_address)
+        self.__output_format_bool(test_case, ret)
+
+        self.__ref.leave()
+        self.__dut.leave()
+
     #
     # Private methods
     #
+    def __run_link_metrics_test_commands(self, initiator: OTCI, subject_address: Ip6Addr) -> bool:
+        seriesid = 1
+        series_flags = 'ldra'
+        link_metrics_flags = 'qr'
+        probe_lenght = 10
+
+        ret = initiator.linkmetrics_config_enhanced_ack_register(subject_address, link_metrics_flags)
+        if not ret:
+            return False
+
+        ret = initiator.linkmetrics_config_forward(subject_address, seriesid, series_flags, link_metrics_flags)
+        if not ret:
+            return False
+
+        initiator.linkmetrics_probe(subject_address, seriesid, probe_lenght)
+
+        results = initiator.linkmetrics_request_single(subject_address, link_metrics_flags)
+        ret = 'lqi' in results.keys() and 'rssi' in results.keys()
+        if not ret:
+            return False
```

Comment:
```suggestion
        if not ('lqi' in results.keys() and 'rssi' in results.keys()):
          return False
```
I feel `ret` is not necessary in this function and all the three lines blocks can be changed to two lines.